File: app/src/main/java/lee/jaebeom/WikiPortal/setting/SettingContract.java
package lee.jaebeom.WikiPortal.setting;

import android.content.SharedPreferences;

import lee.jaebeom.WikiPortal.BasePresenter;
import lee.jaebeom.WikiPortal.BaseView;

/**
 * Created by leejaebeom on 2017. 10. 18..
 */

public interface SettingContract {
    interface View extends BaseView<Presenter>{

    }

    interface Presenter extends BasePresenter{
        void setUsed(SharedPreferences preferences, String key);
        boolean isConfirm();
    }
}

File: app/src/main/java/lee/jaebeom/WikiPortal/setting/SettingContract.kt
package lee.jaebeom.WikiPortal.setting

import android.content.SharedPreferences
import lee.jaebeom.WikiPortal.BasePresenter
import lee.jaebeom.WikiPortal.BaseView
import lee.jaebeom.WikiPortal.util.SharedPreference

/**
 * Created by leejaebeom on 2017. 11. 7..
 */
interface SettingContract {
    interface  View : BaseView<Presenter>{

    }

    interface Presenter : BasePresenter {
        fun setUsed(preferences: SharedPreferences, key: String)
        fun isConfirm() : Boolean
    }
}